### PR TITLE
chore: remove rowpartition/rowoffset feature flag

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

The ROWPARTITION/ROWOFFSET feature has been live for almost a year now and no bugs have been reported. This PR removes the feature flag that controls whether the feature is enabled or not, in order to clean up unused code. 

### Testing done 

Existing tests continue to pass.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/query/QueryExecutorTest.java [hunk 1]
```diff
@@ -187,6 +188,7 @@ public void setUp() {
         ImmutableSet.of(new QueryId("a")));
     when(pushQueryQuery.getRefinement())
         .thenReturn(Optional.of(RefinementInfo.of(OutputRefinement.CHANGES)));
+    when(pushQueryQuery.getSelect()).thenReturn(new Select(Collections.emptyList()));
```
Comment (nateab) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/query/QueryExecutorTest.java:191]: is this related to rowpartition/rowoffset?

**Review by nateab — APPROVED** [review]
LGTM just one minor question

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/query/QueryExecutorTest.java (hunk 1 above)
Comment (vcrfxia) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/query/QueryExecutorTest.java:191]: Indirectly, yes. Because this unit test uses a mocked KsqlConfig, previously the rowpartition/rowoffset feature flag defaulted to false. As a result, the push query validation [here](https://github.com/confluentinc/ksql/blob/f849bc92c6d0f60ec017947afa91da4a31d9967e/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/util/ScalablePushUtil.java#L95) was never actually executed since the feature flag short-circuited the check early. Now that the feature flag is removed, the validation is executed, and it will NPE if the push query does not contain a `Select`.

**Comment by vcrfxia** [comment]
Thanks for the super quick review, @nateab ! 🙏
